File: 53. Maximum Subarray/maximumSubarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        //Make your two variables max_sum and current_sum
        int max_sum = nums[0];
        //Max sum has to be the first number in the array, not 0, because what would happen if they array is a singular negative number?
        //Current sum can 0 or anything below that
        int current_sum = 0;
        
        //Iterate through the nums array, use Kadane's algorithm to find the max sum 
        for(int i = 0; i<nums.length; i++){
            current_sum = Math.max(current_sum + nums[i], nums[i]);
            max_sum = Math.max(current_sum, max_sum);
        }
        
        return max_sum;
        
    }
}

File: 53. Maximum Subarray/maximumSubarray.c
int maxSubArray(int* nums, int numsSize){
    //Make your two variables with max_sum equal to the first number and
    //current_sum equal to 0
    int max_sum = nums[0];
    int current_sum = 0;

    //Iterate through the nums array, use Kadane's algorithm to find the max sum 
    for (int i = 0; i<numsSize; i++){
        if(current_sum + nums[i]> nums[i]) current_sum = current_sum + nums[i];
        else
            current_sum = nums[i];
        if (max_sum < current_sum) max_sum = current_sum;
        else
            max_sum = max_sum;

}
    return max_sum;
}

